code/modules/mob/living/silicon/robot/component.dm

```diff
@@ -107,6 +117,13 @@
 	external_type = /obj/item/robot_parts/robot_component/diagnosis_unit
 	max_damage = 30
 
+//NEW INTEGRATED LIGHT COMPONENT
+/datum/robot_component/integrated_light
+	name = "integrated light"
+	energy_consumption = 4
+	external_type = /obj/item/robot_parts/robot_component/integrated_light
+	max_damage = 40
```

Comment:
No more ridiculous than the precedent that's already in place concerning cameras and radios.